### PR TITLE
fix: change the encoding of actor mapping in grpc

### PR DESCRIPTION
**Pull request by xx01cyx**
## What's changed and what's your intention?

- Summarize your change 
   - Change the encoding of actor mapping in grpc. Compress the data to avoid a too-long grpc message.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

#1443

### CODE REVIEW

File: src/stream/src/task/stream_manager.rs [hunk 1]
```diff
@@ -416,15 +417,15 @@ impl LocalStreamManagerCore {
...
-                    let hash_mapping = dispatcher
-                        .hash_mapping
-                        .clone()
-                        .ok_or_else(|| {
-                            RwError::from(ErrorCode::InternalError(
-                                "hash dispatcher doesn't have consistent hash mapping".to_string(),
-                            ))
-                        })?
-                        .hash_mapping;
+                    let compressed_mapping = dispatcher.hash_mapping.clone().ok_or_else(|| {
```
Comment (BugenZhao) [inline src/stream/src/task/stream_manager.rs:420]: We can get the reference with `get_xx`.
```suggestion
                    let compressed_mapping = dispatcher.get_hash_mapping()?;
```

File: src/common/src/util/compress.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    if let Some(&last) = original_data.last() {
+        original_indices.push(original_data.len() as u64 - 1);
+        data.push(last);
+    }
+
+    (original_indices, data)
+}
+
+/// Works in a reversed way as `compress_data`.
+pub fn decompress_data<T>(original_indices: Vec<u64>, data: Vec<T>) -> Vec<T>
```
Comment (BugenZhao) [inline src/common/src/util/compress.rs:43]: Seems we only need the slice here?
```suggestion
pub fn decompress_data<T>(original_indices: &[u64], data: &[T]) -> Vec<T>
```

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM.

**Review by skyzh — APPROVED** [review]
LGTM, but I'd prefer to store `offsets` in the mapping, so that things might be clearer, and follows most of the developers' conventions. e.g., if we have:

`1, 1, 1, 2, 2, 2, 3, 3, 3`,
then we store `0, 3, 6` in `offsets`, and `1, 2, 3` in data.

**Comment by BugenZhao** [comment]
> LGTM, but I'd prefer to store `offsets` in the mapping, so that things might be clearer, and follows most of the developers' conventions. e.g., if we have:
> 
> `1, 1, 1, 2, 2, 2, 3, 3, 3`, then we store `0, 3, 6` in `offsets`, and `1, 2, 3` in data.

Then we need another field for the length. 😁

**Review by BugenZhao — APPROVED** [review]

**Comment by skyzh** [comment]
> > LGTM, but I'd prefer to store `offsets` in the mapping, so that things might be clearer, and follows most of the developers' conventions. e.g., if we have:
> 
> > 
> 
> > `1, 1, 1, 2, 2, 2, 3, 3, 3`, then we store `0, 3, 6` in `offsets`, and `1, 2, 3` in data.
> 
> 
> 
> Then we need another field for the length. 😁

We can record n+1 offsets 🤪

**Comment by xx01cyx** [comment]
> We can record n+1 offsets 🤪

Then there will be little difference compared with current encoding 🤣

**Comment by skyzh** [comment]
Guess 65536 is still too much to process, it seems that e2e timed out?

**Comment by xx01cyx** [comment]
> Guess 65536 is still too much to process, it seems that e2e timed out?

Might find out the reason later 🤣

**Pull request merged**